src/derivatives.jl

```diff
@@ -1,4 +1,7 @@
-derivative(A, t) = derivative(A, t, firstindex(A.t) - 1)[1]
+function derivative(A, t)
+    ((t < A.t[1] || t > A.t[end]) && !A.extrapolate) && error("Cannot extrapolate!")
```

Comment:
we should make this an error type and throw it.

Reply:
Addressed in [d45c36f](https://github.com/SciML/DataInterpolations.jl/pull/193/commits/d45c36f924c67c9860b4ffea4eb5262df554bd3a)